### PR TITLE
Remove Symfony 4.4 deprecations

### LINKED ISSUE #24605: Remove Symfony 4.4 deprecations
Migrating to Symfony 4.4 introduced new deprecations, we should get rid of them

### PR DESCRIPTION
**Pull request by atomiix**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Upgrading Symfony introduced new deprecations, this aims to get rid of them.
| Type?             | improvement
| Category?         | CO
| BC breaks?        | yes
| Deprecations?     | yes
| Fixed ticket?     | Fixes #24605
| How to test?      | Tests should pass and the whole test suite has to be executed on this PR.
| Possible impacts? | 

## BC Breaks

- The syntax `bundlename:path:file.html.twig` is now invalid, we should use `@bundlename/path/to/file.html.twig`
- Syntax for `_controller` should not use the short notation anymore (`PrestaShopBundle:Admin/Error:enableDebugMode` for instance) it should use the full path: `PrestaShopBundle\Controller\Admin\ErrorController::enableDebugModeAction`
- `MailTemplateTwigRenderer` constructor now takes `Twig\Environment` instead of `Symfony\Component\Templating\EngineInterface`.
- guzzle.cache service is not available anymore
- The following classes are not available anymore:
  - `ModuleTemplateCacheWarmer`
  - `OverrideTwigServiceCompilerPass`
  - `SqlRequestFormDataProvider`
  - `SqlRequestFormHandler`
  - `ModuleTemplateIterator`
- `AppendConfigurationFileHooksListCommand` constructor now takes 9 parameters:
  - `string $env`
  - `LegacyContext $legacyContext`
  - `GridDefinitionHookByServiceIdsProvider $gridDefinitionHookByServiceIdsProvider`
  - `IdentifiableObjectHookByFormTypeProvider $identifiableObjectHookByFormTypeProvider`
  - `HookDescriptionGenerator $hookDescriptionGenerator`
  - `array $serviceIds`
  - `array $optionFormHookNames`
  - `array $formTypes`
  - `string $hookFile`
- `AppendHooksListForSqlUpgradeFileCommand` constructor now takes 10 parameters:
  - `string $env`
  - `LegacyContext $legacyContext`
  - `GridDefinitionHookByServiceIdsProvider $gridDefinitionHookByServiceIdsProvider`
  - `IdentifiableObjectHookByFormTypeProvider $identifiableObjectHookByFormTypeProvider`
  - `HookInformationProvider $hookInformationProvider`
  - `HookDescriptionGenerator $hookDescriptionGenerator`
  - `array $serviceIds`
  - `array $optionFormHookNames`
  - `array $formTypes`
  - `string $sqlUpgradePath`
- `CheckTranslationDuplicatesCommand` constructor now takes 1 parameter:
  - `TranslatorBagInterface $translator`
- `ExportThemeCommand` constructor now takes 3 parameters:
  - `ThemeRepository $themeRepository`
  - `ThemeExporter $themeExporter`
  - `TranslatorInterface $translator`
- `GenerateMailTemplatesCommand` constructor now takes 2 parameters:
  - `CommandBusInterface $commandBus`
  - `LegacyContext $legacyContext`
- `LegacyLinkLinterCommand` constructor now takes 2 parameters:
  - `LegacyLinkLinter $legacyLinkLinter`
  - `AdminRouteProvider $adminRouteProvider`
- `ListCommandsAndQueriesCommand` constructor now takes 2 parameters:
  - `CommandDefinitionParser $commandDefinitionParser`
  - `array $commandAndQueries`
- `ModuleCommand` constructor now takes 4 parameters:
  - `TranslatorInterface $translator`
  - `LegacyContext $context`
  - `ModuleSelfConfigurator $moduleSelfConfigurator`
  - `ModuleManager $moduleManager`
- `NamingConventionLinterCommand` constructor now takes 2 parameters:
  - `AdminRouteProvider $adminRouteProvider`
  - `NamingConventionLinter $namingConventionLinter`
- `SecurityAnnotationLinterCommand` constructor now takes 2 parameters:
  - `AdminRouteProvider $adminRouteProvider`
  - `SecurityAnnotationLinter $securityAnnotationLinter`
- `ThemeEnablerCommand` constructor now takes 1 parameter:
  - `ThemeManager $themeManager`
- `UpdateEUTaxruleGroupsCommand` constructor now takes 1 parameter:
  - `string $localizationPath`
- `UpdateSchemaCommand` constructor now takes 3 parameters:
  - `string $databaseName`
  - `string $databasePrefix`
  - `EntityManager $manager`

## Deprecations

- ~~Template files moved from `src/PrestaShopBundle/Resources/views/` to `templates/bundles/PrestaShopBundle/`~~ #25908
- Translation files moved from `app/Resources/translations/` to `translations/`
- `@AdminSecurity is_granted()` should not take an array, we should use explicit `&&` or `||` when checking multiple rights
- [Form type extensions should stop implementing `public function getExtendedType()` and start implementing `public static function getExtendedTypes(): iterable`](https://symfony.com/blog/new-in-symfony-4-2-improved-form-type-extensions)

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25052)

### CODE REVIEW

File: src/Adapter/MailTemplate/MailTemplateTwigRenderer.php [hunk 1]
```diff
@@ -57,18 +58,18 @@ class MailTemplateTwigRenderer implements MailTemplateRendererInterface
...
-     * @param EngineInterface $engine
+     * @param Environment $twig
      * @param LayoutVariablesBuilderInterface $variablesBuilder
      * @param HookDispatcherInterface $hookDispatcher
      *
      * @throws TypeException
      */
     public function __construct(
-        EngineInterface $engine,
+        Environment $twig,
```
Comment (Progi1984) [inline src/Adapter/MailTemplate/MailTemplateTwigRenderer.php:68]: BC Break : please document

File: src/PrestaShopBundle/Cache/ModuleTemplateCacheWarmer.php [hunk 2]
```diff
@@ -1,114 +0,0 @@
...
-use Symfony\Bundle\FrameworkBundle\CacheWarmer\TemplateFinderInterface;
-use Symfony\Bundle\TwigBundle\CacheWarmer\TemplateCacheCacheWarmer;
-use Symfony\Component\Finder\Finder;
-use Symfony\Component\Templating\TemplateReference;
-use Twig\Error\Error;
-
-/**
- * Generates the Twig cache for all paths, with a specific filter for given namespace
- */
-class ModuleTemplateCacheWarmer extends TemplateCacheCacheWarmer
```
Comment (Progi1984) [inline src/PrestaShopBundle/Cache/ModuleTemplateCacheWarmer.php:39]: BC Break : please document

File: src/PrestaShopBundle/DependencyInjection/Compiler/OverrideTwigServiceCompilerPass.php [hunk 3]
```diff
@@ -1,51 +0,0 @@
...
-use Symfony\Component\DependencyInjection\ContainerBuilder;
-
-/**
- * Twig templates are allowed in the whole modules folder.
- * However, the generated cache parse all the files found each folder in `twig.paths`,
- * including php, css, js, md files and so on.
- *
- * This compiler pass updates the called class to filter the allowed extensions.
- */
-class OverrideTwigServiceCompilerPass implements CompilerPassInterface
```
Comment (Progi1984) [inline src/PrestaShopBundle/DependencyInjection/Compiler/OverrideTwigServiceCompilerPass.php:41]: BC Break

File: src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/RequestSql/SqlRequestFormDataProvider.php [hunk 4]
```diff
@@ -1,247 +0,0 @@
...
-    ),
-    E_USER_DEPRECATED
-);
-
-/**
- * Class RequestSqlFormDataProvider is responsible for getting/saving RequestSql form data.
- *
- * @deprecated Since 1.7.7.5 and will be removed in the next major.
- */
-class SqlRequestFormDataProvider
```
Comment (Progi1984) [inline src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/RequestSql/SqlRequestFormDataProvider.php:54]: BC Break

File: src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/RequestSql/SqlRequestFormHandler.php [hunk 5]
```diff
@@ -1,135 +0,0 @@
...
-    ),
-    E_USER_DEPRECATED
-);
-
-/**
- * Class RequestSqlFormHandler is responsible for creating RequestSql form.
- *
- * @deprecated Since 1.7.7.5 and will be removed in the next major.
- */
-class SqlRequestFormHandler
```
Comment (Progi1984) [inline src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/RequestSql/SqlRequestFormHandler.php:46]: BC Break

**Comment by Julievrz** [comment]
@atomiix I don't think this PR contains any new wording, do you confirm? ;-)

File: src/PrestaShopBundle/Twig/Locator/ModuleTemplateIterator.php [hunk 6]
```diff
@@ -1,109 +0,0 @@
...
-
-use Symfony\Bundle\TwigBundle\TemplateIterator;
-use Symfony\Component\Finder\Finder;
-use Symfony\Component\HttpKernel\KernelInterface;
-
-/**
- * Extended class: Iterator for all templates in bundles and in the application Resources directory.
- * This one applies a specific filter on the module folder, in order to take only *.twig
- */
-class ModuleTemplateIterator extends TemplateIterator
```
Comment (Progi1984) [inline src/PrestaShopBundle/Twig/Locator/ModuleTemplateIterator.php:38]: BC Break

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks (but it's a good job)

**Comment by kpodemski** [comment]
Hey @atomiix 

This is pretty serious BC Break:

> Template files moved from src/PrestaShopBundle/Resources/views/ to templates/bundles/PrestaShopBundle/

Can we make sure that view overrides from modules, are still working as expected with a proper fallback?

**Comment by atomiix** [comment]
@kpodemski I confirm that template overrides are still working ;)

File: src/PrestaShopBundle/Command/CheckTranslationDuplicatesCommand.php [hunk 7]
```diff
@@ -27,13 +27,25 @@
...
 
-class CheckTranslationDuplicatesCommand extends ContainerAwareCommand
+class CheckTranslationDuplicatesCommand extends Command
 {
+    /**
+     * @var TranslatorBagInterface
+     */
+    private $translator;
+
+    public function __construct(TranslatorBagInterface $translator)
```
Comment (Progi1984) [inline src/PrestaShopBundle/Command/CheckTranslationDuplicatesCommand.php:43]: Please document BC breaks in constructor

File: src/PrestaShopBundle/Resources/config/services/bundle/cache.yml [hunk 8]
```diff
@@ -7,13 +7,6 @@ services:
         arguments:
             - "%kernel.cache_dir%/doctrine"
 
-    guzzle.cache:
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/config/services/bundle/cache.yml:10]: BC Break : service not anymore available

**Review by Progi1984 — COMMENTED** [review]
Some comments

File: src/PrestaShopBundle/Command/CheckTranslationDuplicatesCommand.php (hunk 7 above)
Comment (atomiix) [reply src/PrestaShopBundle/Command/CheckTranslationDuplicatesCommand.php:43]: I already documented every command constructor now takes parameters, should I document them individually?

Comment (Progi1984) [reply src/PrestaShopBundle/Command/CheckTranslationDuplicatesCommand.php:43]: If you have a little time, yes, it would be better.

**Review by Progi1984 — DISMISSED** [review]

File: src/Adapter/Module/AdminModuleDataProvider.php [hunk 9]
```diff
@@ -37,7 +37,7 @@
 use PrestaShopBundle\Service\DataProvider\Admin\ModuleInterface;
 use Psr\Log\LoggerInterface;
 use Symfony\Component\Routing\Router;
-use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Contracts\Translation\LocaleAwareInterface;
```
Comment (Progi1984) [inline src/Adapter/Module/AdminModuleDataProvider.php]: Why not use use Symfony\Contracts\Translation\TranslatorInterface ?

File: src/Adapter/Module/ModuleDataProvider.php [hunk 10]
```diff
@@ -52,7 +52,7 @@ class ModuleDataProvider
     /**
      * Translator.
      *
-     * @var \Symfony\Component\Translation\TranslatorInterface
+     * @var \Symfony\Contracts\Translation\TranslatorInterface
```
Comment (Progi1984) [inline src/Adapter/Module/ModuleDataProvider.php]:
```suggestion
     * @var TranslatorInterface
```

File: src/Adapter/Module/Tab/ModuleTabRegister.php [hunk 11]
```diff
@@ -67,7 +67,7 @@ class ModuleTabRegister
     private $logger;
 
     /**
-     * @var \Symfony\Component\Translation\TranslatorInterface
+     * @var \Symfony\Contracts\Translation\TranslatorInterface
```
Comment (Progi1984) [inline src/Adapter/Module/Tab/ModuleTabRegister.php]:
```suggestion
     * @var TranslatorInterface
```

File: src/Adapter/PDF/CreditSlipPdfGenerator.php [hunk 12]
```diff
@@ -59,16 +53,13 @@ final class CreditSlipPdfGenerator implements PDFGeneratorInterface
     private $connection;
 
     /**
-     * @param TranslatorInterface $translator
      * @param string $dbPrefix
      * @param Connection $connection
      */
     public function __construct(
```
Comment (Progi1984) [inline src/Adapter/PDF/CreditSlipPdfGenerator.php]: BC Break

File: src/Core/Addon/Module/ModuleManager.php [hunk 13]
```diff
@@ -79,7 +79,7 @@ class ModuleManager implements AddonManagerInterface
     /**
      * Translator.
      *
-     * @var \Symfony\Component\Translation\TranslatorInterface
+     * @var \Symfony\Contracts\Translation\TranslatorInterface
```
Comment (Progi1984) [inline src/Core/Addon/Module/ModuleManager.php]:
```suggestion
     * @var TranslatorInterface
```

File: src/Core/Addon/Module/ModuleRepository.php [hunk 14]
```diff
@@ -42,8 +41,10 @@
 use PrestaShop\PrestaShop\Core\Addon\AddonListFilterType;
 use PrestaShop\PrestaShop\Core\Addon\AddonsCollection;
 use Psr\Log\LoggerInterface;
+use Symfony\Component\Cache\Adapter\ArrayAdapter;
+use Symfony\Component\Cache\DoctrineProvider;
 use Symfony\Component\Finder\Finder;
-use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
```
Comment (Progi1984) [inline src/Core/Addon/Module/ModuleRepository.php]: PHPStan :warning:

File: src/Core/Addon/Theme/ThemeValidator.php [hunk 15]
```diff
@@ -27,14 +27,14 @@
...
-use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
 
 class ThemeValidator
 {
     /**
      * Translator.
      *
-     * @var \Symfony\Component\Translation\TranslatorInterface
+     * @var \Symfony\Contracts\Translation\TranslatorInterface
```
Comment (Progi1984) [inline src/Core/Addon/Theme/ThemeValidator.php]:
```suggestion
     * @var TranslatorInterface
```

File: src/PrestaShopBundle/Resources/config/services/core/circuit_breaker.yml [hunk 16]
```diff
@@ -3,11 +3,18 @@ services:
   _defaults:
     public: true
 
-  prestashop.core.circuit_breaker.doctrine_cache:
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/config/services/core/circuit_breaker.yml:6]: Service not anymore available

**Review by PierreRambaud — DISMISSED** [review]
I think it's time to do a first merge and continue to another PR, it becomes unreviewable :sweat_smile:

**Review by Progi1984 — COMMENTED** [review]
Some easy comments

**Comment by atomiix** [comment]
@PierreRambaud yeah, that's what I started to think, I'll do the `translatorInterface` in another PR.

File: src/PrestaShopBundle/Resources/config/services/core/circuit_breaker.yml (hunk 16 above)
Comment (atomiix) [reply src/PrestaShopBundle/Resources/config/services/core/circuit_breaker.yml:6]: It is, [just below](https://github.com/PrestaShop/PrestaShop/pull/25052/files#diff-5a786fa5a57e5490a73db3e5b6bd561ad20f26af29cc2d1477ec57b7854fa804R6)

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

File: app/config/routing_dev.yml [hunk 17]
```diff
@@ -12,31 +12,3 @@ _errors:
 
 _main:
     resource: routing.yml
-
-_admin_stock_hot_reload_json:
```
Comment (jolelievre) [inline app/config/routing_dev.yml:16]: Does it mean these routes don't exist any more? (I couldn't find a place that were moved to)
They are not used any more?
Isn't it a BC break?

File: src/PrestaShopBundle/Command/GenerateMailTemplatesCommand.php [hunk 18]
```diff
@@ -86,23 +103,19 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
     {
-        /** @var LegacyContext $legacyContext */
-        $legacyContext = $this->getContainer()->get('prestashop.adapter.legacy.context');
         //We need to have an employee or the module hooks don't work
         //see LegacyHookSubscriber
-        if (!$legacyContext->getContext()->employee) {
+        if (!$this->legacyContext->getContext()->employee) {
             //Even a non existing employee is fine
-            $legacyContext->getContext()->employee = new Employee(42);
+            $this->legacyContext->getContext()->employee = new Employee(42);
```
Comment (jolelievre) [inline src/PrestaShopBundle/Command/GenerateMailTemplatesCommand.php:118]:
```suggestion
            $this->legacyContext->getContext()->employee = new Employee();
```

Comment (jolelievre) [reply src/PrestaShopBundle/Command/GenerateMailTemplatesCommand.php:118]: Maybe this would be cleaner, less prone to error?

File: src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/RequestSql/SqlRequestFormDataProvider.php [hunk 19]
```diff
@@ -1,247 +0,0 @@
...
-
-@trigger_error(
-    sprintf(
-        '%s is deprecated since version 1.7.7.5 and will be removed in the next major version.',
-        SqlRequestFormDataProvider::class
-    ),
-    E_USER_DEPRECATED
-);
-
-/**
```
Comment (jolelievre) [inline src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/RequestSql/SqlRequestFormDataProvider.php:49]: Why did you remove this one? Is it related to the Symfony upgrade?
Or is it because it was marked as a deprecated component that should be removed?

File: src/PrestaShopBundle/Form/Admin/Extension/AlertExtension.php [hunk 20]
```diff
@@ -80,8 +80,8 @@ public function buildView(FormView $view, FormInterface $form, array $options)
     /**
      * {@inheritdoc}
      */
-    public function getExtendedType()
+    public static function getExtendedTypes(): iterable
```
Comment (jolelievre) [inline src/PrestaShopBundle/Form/Admin/Extension/AlertExtension.php:83]: Ooohh so we can extend multiple types now!! That's nice, I hop the service definition doesn't require to repeat this info anymore (it used to be duplicated in both the extension and the service configuration)

https://github.com/PrestaShop/PrestaShop/blob/b0749a4de9f91c5cd154d7d9c0c5b9266825267c/src/PrestaShopBundle/Resources/config/services/bundle/form/form_extension.yml#L48

Comment (jolelievre) [reply src/PrestaShopBundle/Form/Admin/Extension/AlertExtension.php:83]: Maybe you should indicate this is the BC break descriptions, that all the form extensions now implement this static method

File: src/PrestaShopBundle/Resources/config/routing/admin/configure/advanced_parameters/logs.yml [hunk 21]
```diff
@@ -2,29 +2,29 @@ admin_logs_index:
     path: /
     methods: [GET]
     defaults:
-        _controller: 'PrestaShopBundle:Admin\Configure\AdvancedParameters\Logs:index'
+        _controller: 'PrestaShopBundle\Controller\Admin\Configure\AdvancedParameters\LogsController::indexAction'
```
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/config/routing/admin/configure/advanced_parameters/logs.yml:5]: I wonder, since this is not allowed anymore in version 5.0

Is there maybe a configuration in the framework the enabled/disabled the use of the short notation? If so maybe we should force it to avoid us adding short notation again?

But maybe this would also force modules to update their routing configuration then... so maybe not such a good idea for now.

File: src/PrestaShopBundle/Resources/config/services/bundle/form/form_handler.yml [hunk 22]
```diff
@@ -495,22 +495,6 @@ services:
             - 'PrestaShopBundle\Form\Admin\Improve\Design\Theme\ShopLogosType'
             - 'ShopLogo'
 
-    # Entity form handler
-    prestashop.admin.request_sql.form_handler:
```
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/config/services/bundle/form/form_handler.yml:499]: Same question as before, but why remove these services?

**Review by jolelievre — COMMENTED** [review]
Still reviewing (244 / 932) but here a re a few comments/questions

File: templates/bundles/PrestaShopBundle/Admin/Configure/AdvancedParameters/Email/Blocks/email_configuration.html.twig [hunk 23]
```diff
@@ -42,8 +42,8 @@
           {{ form_row(emailConfigurationForm.log_emails) }}
           <div class="alert alert-info" role="alert">
             <p class="alert-text">
-              {{ 'Fill in the fields below to set up DKIM signing of outgoing emails. This will reduce the likelihood of your emails being marked as spam.
```
Comment (jolelievre) [inline templates/bundles/PrestaShopBundle/Admin/Configure/AdvancedParameters/Email/Blocks/email_configuration.html.twig:45]: Isn't this gonna change the wording key?

File: app/config/routing_dev.yml (hunk 17 above)
Comment (PierreRambaud) [reply app/config/routing_dev.yml:16]: We accept this kind of BC, and hot reload is not used anymore

Comment (atomiix) [reply app/config/routing_dev.yml:16]: Yeah, it's been a long time since those routes are not used anymore, the actions associated don't even exist 😅

File: src/PrestaShopBundle/Form/Admin/Extension/AlertExtension.php (hunk 20 above)
Comment (PierreRambaud) [reply src/PrestaShopBundle/Form/Admin/Extension/AlertExtension.php:83]: It's a BC break from Symfony, not sure we have to document them all

File: templates/bundles/PrestaShopBundle/Admin/macros.html.twig [hunk 24]
```diff
@@ -208,7 +208,7 @@
           data-trigger="hover"
           data-container=".{{ popoverContainer }}"
         >
-          <i class="material-icons form-error-icon">error_outline</i> <u>{{ '%count% errors'|transchoice(popoverErrors|length, {}, 'Admin.Global') }}</u>
+          <i class="material-icons form-error-icon">error_outline</i> <u>{{ '%count% errors'|trans({'%count%': popoverErrors|length}, 'Admin.Global') }}</u>
```
Comment (jolelievre) [inline templates/bundles/PrestaShopBundle/Admin/macros.html.twig:211]: Does it mean `transchoice` cannot be used anymore?

File: src/PrestaShopBundle/Resources/config/services/bundle/form/form_handler.yml (hunk 22 above)
Comment (PierreRambaud) [reply src/PrestaShopBundle/Resources/config/services/bundle/form/form_handler.yml:499]: Service has been marked as deprecated in the previous version.
It's time to remove them now :D

**Review by PierreRambaud — APPROVED** [review]

**Review by jolelievre — COMMENTED** [review]
Las comments for the whole PR
I lost hope checking all the files as Viewed 😅. Especially for those that were just renamed

File: templates/bundles/PrestaShopBundle/Admin/macros.html.twig (hunk 24 above)
Comment (atomiix) [reply templates/bundles/PrestaShopBundle/Admin/macros.html.twig:211]: It is deprecated 😄

File: src/PrestaShopBundle/Form/Admin/Extension/AlertExtension.php (hunk 20 above)
Comment (jolelievre) [reply src/PrestaShopBundle/Form/Admin/Extension/AlertExtension.php:83]: Not all but just indicate in the PR description that the extensions must now respect this new interface

File: src/PrestaShopBundle/Resources/config/services/bundle/form/form_handler.yml (hunk 22 above)
Comment (jolelievre) [reply src/PrestaShopBundle/Resources/config/services/bundle/form/form_handler.yml:499]: Yes but it's not the place of this PR, this PR is about deprecations related to the Symfony upgrade not our own
Our deprecated services/classes should be removed in a dedicated PR
(but I'm not gonna block for this one of course)

File: templates/bundles/PrestaShopBundle/Admin/Configure/AdvancedParameters/Email/Blocks/email_configuration.html.twig (hunk 23 above)
Comment (atomiix) [reply templates/bundles/PrestaShopBundle/Admin/Configure/AdvancedParameters/Email/Blocks/email_configuration.html.twig:45]: Possibly, yeah... But I guess it's better without the space and we can add the new key, right?

File: src/PrestaShopBundle/Form/Admin/Extension/AlertExtension.php (hunk 20 above)
Comment (atomiix) [reply src/PrestaShopBundle/Form/Admin/Extension/AlertExtension.php:83]: @jolelievre service definition doesn't require to repeat this info anymore ;) https://symfony.com/blog/new-in-symfony-4-2-improved-form-type-extensions
I think we can get rid of them in another PR.

I don't think this is a BC break because it's only deprecated. But I can add it to the noticeable changes though!

File: src/PrestaShopBundle/Command/GenerateMailTemplatesCommand.php (hunk 18 above)
Comment (atomiix) [reply src/PrestaShopBundle/Command/GenerateMailTemplatesCommand.php:118]: Well, you're probably right, but maybe we should do it in another PR?

File: src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/RequestSql/SqlRequestFormDataProvider.php (hunk 19 above)
Comment (atomiix) [reply src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/RequestSql/SqlRequestFormDataProvider.php:49]: It was because it was marked as deprecated if I recall 😅

File: src/PrestaShopBundle/Resources/config/services/bundle/form/form_handler.yml (hunk 22 above)
Comment (atomiix) [reply src/PrestaShopBundle/Resources/config/services/bundle/form/form_handler.yml:499]: Same here, I think it was because it was marked as deprecated 😅

**Review by boubkerbribri — APPROVED** [review]
The whole functional campaign passed.
It's QA :heavy_check_mark: 
Thanks @atomiix

File: src/PrestaShopBundle/Form/Admin/Extension/AlertExtension.php (hunk 20 above)
Comment (jolelievre) [reply src/PrestaShopBundle/Form/Admin/Extension/AlertExtension.php:83]: Yep ok for noticeable changes!! Because I'm surprised it didn't cause any bugs, when I go in my Editor I get errors stating that the extension doesn't respect the interface (before your changes)

So it means if modules use form extension they need to adapt to this new way of implementing them.

File: templates/bundles/PrestaShopBundle/Admin/Configure/AdvancedParameters/Email/Blocks/email_configuration.html.twig (hunk 23 above)
Comment (jolelievre) [reply templates/bundles/PrestaShopBundle/Admin/Configure/AdvancedParameters/Email/Blocks/email_configuration.html.twig:45]: Mmm seems like it's a new wording from develop branch so we can change it However the weirdest part here is to have line breaks inside the wording. The wording should be on a single line.

File: templates/bundles/PrestaShopBundle/Admin/macros.html.twig (hunk 24 above)
Comment (jolelievre) [reply templates/bundles/PrestaShopBundle/Admin/macros.html.twig:211]: Yeah that's too bad... We just started introducing 2 or 3 plural messages in 178 😅
It's too bad it's a nice feature to have one wording to rule them all, but anyway..

**Review by jolelievre — APPROVED** [review]
Nothing blocking but some things should maybe be improved in separate PRs:
- wording with line break
- update all form extensions service definitions to remove the extended type
- remove the 42 Employee id in the template generation command

File: templates/bundles/PrestaShopBundle/Admin/macros.html.twig (hunk 24 above)
Comment (Hlavtox) [reply templates/bundles/PrestaShopBundle/Admin/macros.html.twig:211]: @jolelievre @atomiix _The transchoice tag is deprecated since Symfony 4.2 and will be removed in 5.0. Use the ICU MessageFormat with the trans tag instead._

➡️ https://symfony.com/doc/4.4/translation/message_format.html#pluralization
